framework/include/actions/ActionWarehouse.h

```diff
@@ -157,6 +170,28 @@ class ActionWarehouse : public ConsoleStreamInterface
     return action_vector;
   }
 
+  /**
+   * Retrieve all Physics with a specific type ordered by their names
+   */
+  template <class T>
```

Comment:
Why we need to use this function with templates? I could be using this function with a `getPhysics<Real>`, which makes no sense. If template must be preserved, please check that the actual class on which the function is templated makes sense

Reply:
If you use it with the wrong type, it won't find any physics to return

Reply:
I ll add an assert though it should help